### ISSUE TITLE
"P2P Magic Happening"

### ISSUE THREAD

**Issue by arpee** [SaitoTech/saito-lite-rust#283]
Leaving tab open on arcade occasionally results in attached image text showing.

Goes away on refresh and reappears sporadically.

**Comment by trevelyan** [comment]
Comments on Possible Causes:

The arcade uses the following transaction-types to i

OPEN - adds game to arcade
JOIN - contribute sigs so final player can "accept"
ACCEPT - if you have enough sigs
CLOSE - remove game from arcade

We also try to be sophisticated in pushing people into an init-screen as soon as they do something that *should* result in a game initializing. If a player broadcasts an ACCEPT command, for instance, they are pushed into the init screen. There are situations where the ACCEPT command may not result in reciprocal initialization though. The reason for this is that the ACCEPT command just broadcasts the sigs authorizing those players to start the game. The player that created the game may already have received an ACCEPT command from someone else.

So there sorts of edge-cases exist, but should be infrequent. Another example is if a 6 player game has 1 OPEN and then 5 JOIN commands rather than 1 OPEN, 4 JOIN and 1 ACCEPT. This can happen if players receive or do not receive all JOINS for network stability reasons like their websocket goes down. In this situation, the player that sends the OPEN instruction (player creating the game) must create the ACCEPT transaction themselves after receiving 5 JOIN txs.

Basically, there are edge cases. If this is a common problem, the issue is probably that the other player has left or is incommunicado or has already started the game with someone else. Would suggest we kick-off a timer and report error message to user if this screen is still showing in 45 seconds. If it does not happen very frequently it should not be a problem though. I have yet to have this issue on the new arcade, for instance. It may be happening to people in China more frequently because of websocket failures caused by the Great Firewall though.

**Comment by estapeluo** [comment]
This is a common thing to pop out when we start 3-players game (settlers). It shows briefly on the players then goes into init animation. But for people watching the arcade, it shows this page after their game move from arcade->game (star button shows)
